exporter/datadogexporter/internal/logs/sender_test.go

```
@@ -0,0 +1,175 @@
+				counter++
+			},
+		},
+		{
+			name: "does two batches",
+			payload: []datadogV2.HTTPLogItem{{
+				Ddsource:             nil,
+				Ddtags:               datadog.PtrString("tag1:true"),
+				Hostname:             nil,
+				Message:              "",
+				Service:              nil,
+				UnparsedObject:       nil,
+				AdditionalProperties: nil,
+			}, {
+				Ddsource:             nil,
+				Ddtags:               datadog.PtrString("tag1:true"),
+				Hostname:             nil,
+				Message:              "",
+				Service:              nil,
+				UnparsedObject:       nil,
+				AdditionalProperties: nil,
+			}, {
+				Ddsource:             nil,
+				Ddtags:               datadog.PtrString("tag2:true"),
+				Hostname:             nil,
+				Message:              "",
+				Service:              nil,
+				UnparsedObject:       nil,
+				AdditionalProperties: nil,
+			}, {
+				Ddsource:             nil,
+				Ddtags:               datadog.PtrString("tag2:true"),
+				Hostname:             nil,
+				Message:              "",
+				Service:              nil,
+				UnparsedObject:       nil,
+				AdditionalProperties: nil,
+			}},
+			testFn: func(jsonLogs testutil.JsonLogs) {
+				switch counter {
+				case 0:
+					validateTags(t, jsonLogs, "tag1:true")
+					assert.Len(t, jsonLogs, 2)
+				case 1:
+					validateTags(t, jsonLogs, "tag2:true")
+					assert.Len(t, jsonLogs, 2)
+				default:
+					t.Fail()
+				}
+				counter++
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := testutil.DatadogLogServerMock(func() (string, http.HandlerFunc) {
+				return "/api/v2/logs", func(writer http.ResponseWriter, request *http.Request) {
+					jsonLogs := testutil.MockLogsEndpoint(writer, request)
+					tt.testFn(jsonLogs)
+				}
+			})
+			defer server.Close()
+			counter = 0
+			s := NewSender(server.URL, logger, exporterhelper.TimeoutSettings{Timeout: time.Second * 10}, true, true, "")
+			_ = s.SubmitLogs(context.Background(), tt.payload)
```

Don't ignore the error please.